azurelinuxagent/ga/monitor.py

```diff
@@ -247,10 +249,14 @@ def collect_and_send_events(self):
         """
         Periodically send any events located in the events folder
         """
-        event_list = collect_events()
+        try:
+            event_list = collect_events()
 
-        if len(event_list.events) > 0:
-            self.protocol.report_event(event_list)
+            if len(event_list.events) > 0:
+                self.protocol.report_event(event_list)
+        except Exception as e:
+            err_msg = "Failure in collecting/sending Agent events: {0}".format(textutil.str_to_encoded_ustr(e))
```

Comment:
PeriodicOperation was already handling exceptions for us here, I mainly added the explicit checks here to send out a Kusto event incase the `collect_and_send_events` ever throws some unknown error